web/src/components/settings/LangSelector.vue

```diff
@@ -0,0 +1,51 @@
+<script lang="ts" setup>
+import { computed, ref, toRefs } from "vue"
+import { Language } from "@/services/language"
+import { LanguageWithIndex } from "@/pages/setting/SettingContent.vue"
+
+
+interface Props {
+  langlist: Language[] | LanguageWithIndex[]
+  selectedtab: string
+}
+const props = defineProps<Props>()
+
+const { selectedtab } = toRefs(props);
+
+const selectedOpt = ref(localStorage.getItem(selectedtab.value) ?? 'en')
+
+const updateHandler = (code: string) => {
```

Comment:
Avoid having logic in components. Use props and emits from the parent. 

I don't think we should have code pointing to localStorage and other browser functions multiple places.